### PR TITLE
Feature/10 continuous deployment for gen2 function

### PR DESCRIPTION
**Pull request by Sachinbisht27**
#10 

### Types of changes

- [ ] Bug fix (a change which fixes an issue)
- [x] New feature (a change which adds functionality)

### Short description of what this resolves:

- GitHub workflow for the CD in both environments.
- GitHub workflow for the Database schema migrations.
- Defined the secret variables required for the deployment and migration in GitHub secretes.
- Configured the workflows to run on Push action only.

### Checklist:

- [x] I have performed a self-review of my own code.
- [x] The code follows the style guidelines of this project.
- [x] The code changes are passing the CI checks
- [x] I have documented my code wherever required.
- [ ] The changes requires a change to the documentation.
- [ ] I have updated the documentation based on the my changes.
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.\

### CODE REVIEW

**Review by Satendra-SR — CHANGES_REQUESTED** [review]
@Sachinbisht27 Added a few comments/suggestions. Please have a look.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        FLASK_APP: ${{ secrets.FLASK_APP }}
+        FLASK_ENV: ${{ secrets.FLASK_APP_STAGING }}
+      run: |
+        flask db upgrade
+
+    - name: Stop Cloud SQL Proxy
+      run: |
+        pkill cloud_sql_proxy
+
+    - name: 'Deploy a gen 2 cloud function'
```
[the remark this answers is not in the document]
Comment (Sachinbisht27) [reply .github/workflows/build.yml:60]: Yes, Sure. Let me update it.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+          --entry-point=handle_payload \
+          --memory=256MB \
+          --timeout=30s \
+          --set-env-vars=FLASK_APP=${{ secrets.FLASK_APP_STAGING }},DB_USER=${{ secrets.DB_USER }},DB_PASSWORD=${{ secrets.DB_PASSWORD }},DB_NAME=${{ secrets.DB_NAME }},DB_PORT=${{ secrets.DB_PORT }},CONNECTION_NAME=${{ secrets.CONNECTION_NAME }},LOGGING_LEVEL=${{ secrets.LOGGING_LEVEL }}
+
+  deploy-production:
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - uses: 'actions/checkout@v3'
```
[the remark this answers is not in the document]
Comment (Sachinbisht27) [reply .github/workflows/build.yml:76]: Done

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      - id: 'auth'
+        name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@v1'
```
[the remark this answers is not in the document]
Comment (Sachinbisht27) [reply .github/workflows/build.yml:79]: Updated

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+          credentials_json: '${{ secrets.GCP_CREDENTIALS }}'
+
+      - name: Debug GCP credentials
+        env:
+          GOOGLE_APPLICATION_CREDENTIALS: ${{ secrets.GCP_CREDENTIALS }}
+        run: |
+          echo "$GOOGLE_APPLICATION_CREDENTIALS" > credentials.json
+
+      - name: 'Set up Cloud SDK'
+        uses: 'google-github-actions/setup-gcloud@v1'
```
[the remark this answers is not in the document]
Comment (Sachinbisht27) [reply .github/workflows/build.yml:90]: Updated

File: config.py [hunk 5]
```diff
@@ -32,4 +32,11 @@
         % POSTGRES
     )
 
+github_action_db_url = os.environ.get(
+    "SQLALCHEMY_DATABASE_URI_GITHUB_ACTION", None
+)
+
+if github_action_db_url:
+    SQLALCHEMY_DATABASE_URI = github_action_db_url
+
```
[the remark this answers is not in the document]
Comment (Sachinbisht27) [reply config.py]: Sure @Satendra-SR. Updated the changes.

**Review by Satendra-SR — APPROVED** [review]

**Pull request merged**